features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/presequite/CpuRequirementChecker.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.presequite
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.file.listFilesSafe
+import java.io.File
+
+internal class CpuRequirementChecker(
+    private val cpuDirFile: File = File(DIR_PATH),
+    private val internalLogger: InternalLogger
+) : SystemRequirementChecker {
+
+    private var checkedValue: Int? = null
```

Comment:
shouldn't it be simply `lateinit`? I guess number of cores cannot change over the run of the application?

Reply:
Indeed, But we can't ensure from this class inside that `readCpuCoreNumber` will be correctly called before `checkedValue`, so to avoid crash as possible as we can, I prefer a nullable value here, even if it is null, it will just have a empty value of cpu cores in telemetry.